ballerina/http_response.bal

```diff
@@ -517,6 +517,8 @@ public class Response {
             self.setBodyParts(payload);
         } else if payload is stream<SseEvent, error?> {
             self.setSseEventStream(payload);
+        } else if payload is anydata {
+            self.setJsonPayload(payload.toJson());
```

Comment:
import ballerina/io;

public function main() {
    anydata a = re `a`;
    io:println(a.toJson());
}

If this kind of a scenario, it will throws a runtime error of 
```
error: {ballerina/lang.value}ConversionError {"message":"'anydata & readonly' value cannot be converted to 'json'"}
        at ballerina.lang.value.0:toJson(value.bal:262)
           admin.issue_6926.0:main(main.bal:6)
```

Reply:
You mean it will panic right? Shouldn't we handle this in the lang level. The `toJson` returns a `json` not `json|error`.

Reply:
+1, can you create a issue for this, As a work around can you create a type check for this `if payload is string:RegExp`

Reply:
I think this is a very rare case (providing a regular expression in the `setPayload` method). So I don't think we need to explicitly handle this. Since this is a lang issue, if they decided to fix it by giving an error or string value of the regular expression, we will change accordingly in the future. 

Created an issue here: https://github.com/ballerina-platform/ballerina-lang/issues/43360